samples/bluetooth/hci_vs_scan_req/src/main.c

```diff
@@ -0,0 +1,228 @@
+/* main.c - Application main entry point */
+
+/*
+ * Copyright (c) 2024 Giancarlo Stasi
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/bluetooth/hci_vs.h>
+#include <zephyr/bluetooth/addr.h>
+
+#define DEVICE_NAME CONFIG_BT_DEVICE_NAME
+#define DEVICE_NAME_LENGTH (sizeof(DEVICE_NAME) - 1)
+
+/* Advertising Interval: the longer, the less energy consumption.
+ * Units: 0.625 milliseconds.
+ * The Minimum Advertising Interval and Maximum Advertising Interval should not be the same value
+ * (as stated in Bluetooth Core Spec 5.2, section 7.8.5)
+ */
+#define ADV_MIN_INTERVAL        BT_GAP_ADV_SLOW_INT_MIN
+#define ADV_MAX_INTERVAL        BT_GAP_ADV_SLOW_INT_MAX
+
+#if defined(CONFIG_BT_PERIPHERAL)
+#define ADV_OPTIONS             (BT_LE_ADV_OPT_CONNECTABLE | BT_LE_ADV_OPT_SCANNABLE | \
+				 BT_LE_ADV_OPT_NOTIFY_SCAN_REQ)
+#else
+#define ADV_OPTIONS             (BT_LE_ADV_OPT_SCANNABLE | BT_LE_ADV_OPT_NOTIFY_SCAN_REQ)
+#endif
+
+/*
+ * Radio output power for advertisements. Lower the value, closer the scanner.
+ * Values are defined in HAL (e.g. modules/hal/nordic/nrfx/mdk/nrf52_bitfields.h)
+ * signed 8-bit representing dBm.
+ */
+#define ADV_TX_POWER ((int8_t)(-8))
+
+static uint8_t scan_data[] = {'V', 'S', ' ', 'S', 'a', 'm', 'p', 'l', 'e'};
+
+static const struct bt_le_adv_param parameters = {
+	.options = ADV_OPTIONS,
+	.interval_min = ADV_MIN_INTERVAL,
+	.interval_max = ADV_MAX_INTERVAL,
+};
+
+static const struct bt_data adv_data[] = {
+	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
+	BT_DATA(BT_DATA_NAME_COMPLETE, DEVICE_NAME, DEVICE_NAME_LENGTH),
+};
+
+static const struct bt_data scan_rsp_data[] = {
+	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
+	BT_DATA(BT_DATA_NAME_COMPLETE, scan_data, sizeof(scan_data)),
+};
```

Comment:
IIRC a device may only have single name, so if you use `BT_DATA_NAME_COMPLETE` in both the ADV data and the scan response data, they shall be the same value. 

You can use e.g. `BT_DATA_MANUFACTURER_DATA` if you want to use some advertising data without any requirements

Reply:
Sorry for my lack of fantasy. Hope it's better now.

Reply:
You may have missed something here :) 

My point is that you shouldn't have different values for `BT_DATA_NAME_COMPLETE` in the advertising and scan response data. 

So I suggest to do either

```suggestion
static const struct bt_data scan_rsp_data[] = {
	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
	BT_DATA(BT_DATA_NAME_COMPLETE, DEVICE_NAME, DEVICE_NAME_LENGTH),
};
```

or

```suggestion
static const struct bt_data scan_rsp_data[] = {
	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
	BT_DATA(BT_DATA_MANUFACTURER_DATA, scan_data, sizeof(scan_data)),
};
```

Reply:
ok for the 2nd